ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/package.scala

```diff
@@ -0,0 +1,109 @@
+package com.daml.ledger.participant.state
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.v1.{
+  Configuration,
+  SubmittedTransaction,
+  SubmitterInfo,
+  TransactionMeta,
+  Update
+}
+import com.digitalasset.daml.lf.data.Time.Timestamp
+import com.digitalasset.daml.lf.engine.Engine
+import com.digitalasset.daml_lf.DamlLf.Archive
+
+/** The participant-state key-value utilities provide methods to succintly implement
+  * participant-state on top of ledger's that provide a key-value state storage.
+  *
+  * Key-value utilities are based around the concept of modelling the ledger around
+  * an abstract state that can be described (abstractly) as the tuple
+  * (List[DamlLogEntryId], Map[DamlLogEntryId, DamlLogEntry], Map[DamlStateKey, DamlStateValue]).
```

Comment:
Also explain that if one allocates `DamlLogEntryId`s consecutively, then the list can be stored implicitly via the list of ordered entry-ids.